pkg/agent/core/state.go

```diff
@@ -727,8 +736,20 @@ func (s *state) desiredResourcesFromMetricsOrRequestedUpscaling(now time.Time) (
 	// 2. Cap the goal CU by min/max, etc
 	// 3. that's it!
 
+	reportGoals := func(goalCU uint32, parts scalingevents.GoalCUComponents) {
```

Comment:
Suggestion: Instead of having callback here, we could merge `scalingevents.GoalCUComponents` into `scalingGoal`, define a method `GoalCU()` on it, which would return max, and we can put this object into DesiredScaling callback.

WDYT?

Reply:
Hm, how will that interact with things like #1129 / #1140 ? Otherwise I like this idea, I think it's a lot simpler.